app/src/main/java/com/github/polybooks/FillSaleActivity.kt

```diff
@@ -138,19 +140,15 @@ class FillSaleActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener
      * (i.e. book, condition, user, price, date)
      */
     fun confirmSale(view: View) {
-        // store Sale in our database
-        val sale = Sale(
-            bookFuture.get()!!, // TODO maybe ensure above that the button is disabled if book is null
-            LocalUser, // TODO user
-            findViewById<EditText>(R.id.filled_price).text.toString().toFloat(),
-            // Should never be null as the button is not enabled otherwise
-            bookConditionSelected!!,
-            Timestamp.now(),
-            SaleState.ACTIVE,
-            null
+        salesDB.addSale(
+            bookISBN = bookFuture.get()!!.isbn,
+            seller = LoggedUser(123456, "Alice"), //TODO handle real User
+            price = findViewById<EditText>(R.id.filled_price).text.toString().toFloat(),
```

Comment:
You have to check if the price is >= 0 and maybe also an upper bound to avoid some troll x)